src/plugins/intel_cpu/src/nodes/reorder.cpp

```diff
@@ -417,6 +414,380 @@ void Reorder::reorderData(const IMemory &input, const IMemory &output, MultiCach
+        break;
+    case 1:
+        // Only need pre_converter
+        pre_converter = std::make_shared<IntermConverter>();
+        pre_converter->setInputPrec(src_prc);
+        pre_converter->setOutputPrec(dst_prc);
+        break;
+    case 2:
+        // Only need post_converter
+        post_converter = std::make_shared<IntermConverter>();
+        post_converter->setOutputPrec(dst_prc);
+        post_converter->setInputPrec(src_prc);
+        break;
+    case 3:
+        // Input and output both are not supported
+        pre_converter = std::make_shared<IntermConverter>();
+        pre_converter->setInputPrec(src_prc);
+
+        post_converter = std::make_shared<IntermConverter>();
+        post_converter->setOutputPrec(dst_prc);
+
+        // Choose data type with the smallest data size
+        near_data_type = find_near_data_type(src_prc, dst_prc);
+        OPENVINO_ASSERT(isSupportedDataType(engine, cache, src, near_data_type));
+        pre_converter->setOutputPrec(DnnlExtensionUtils::DataTypeToIEPrecision(near_data_type));
+        post_converter->setInputPrec(DnnlExtensionUtils::DataTypeToIEPrecision(near_data_type));
+        break;
+    default:
+        OPENVINO_THROW("Incorrect converter mask!");
+        break;
+    }
+}
+
+void Reorder::ReorderExecutor::prepareParams(const dnnl::engine& engine,
+                                             MultiCachePtr& cache,
+                                             const ov::intel_cpu::MemoryCPtr& src,
+                                             const ov::intel_cpu::MemoryCPtr& dst,
+                                             const std::vector<int> src_permutation) {
+    src_blocked = std::make_shared<Memory>(engine, src->getDesc(), src->getData(), false);
+    dst_blocked = std::make_shared<Memory>(engine, dst->getDesc(), dst->getData(), false);
+
+    if (pre_converter) {
+        // Set input memory
+        pre_converter->setInputMem(src_blocked);
+
+        // Allocate or resize output memeory
+        auto dst_desc = src->getDesc().cloneWithNewPrecision(pre_converter->getOutputPrec());
+        auto dst_mem = pre_converter->getOutputMem();
+        if (dst_mem) {
+            dst_mem->redefineDesc(dst_desc);
+        } else {
+            auto mem = std::make_shared<Memory>(engine, dst_desc, nullptr, false);
+            pre_converter->setOutputMem(mem);
+        }
+        src_blocked = pre_converter->getOutputMem();
+    }
+
+    if (post_converter) {
+        // Set output memory
+        post_converter->setOutputMem(dst_blocked);
+
+        // Allocate or resize input memeory
+        auto src_desc = dst->getDesc().cloneWithNewPrecision(post_converter->getInputPrec());
+        auto src_mem = post_converter->getInputMem();
+        if (src_mem) {
+            src_mem->redefineDesc(src_desc);
+        } else {
+            auto mem = std::make_shared<Memory>(engine, src_desc, nullptr, false);
+            post_converter->setInputMem(mem);
+        }
```

Comment:
As I mentioned above, it's enough to have just one intermediate storage. Moreover we can use `scratchpad` for this purpose. Thus, we don't need this step at all.

Reply:
It seems we need 2 intermediate storages, for example {input=FP64, output=FP64} oneDNN doesn't support, we need convert to FP32 and then do reorder:

FP64(input) -> convert  -> FP32 -> reorder -> FP32 -> convert to FP64(output)

It needs 2 FP32 memory. Is it right?